### PR TITLE
feat!: fix types for exported entrypoints and simplify others

### PR DESCRIPTION
**Pull request by hugomrdias**
- [fix types for exported entrypoints](https://github.com/ipld/js-dag-ucan/pull/50/commits/a2fd709c6f5b87a9a9e7bad55f766d89f273fcb8)
- [fix: simplify cid,link,proof](https://github.com/ipld/js-dag-ucan/pull/50/commits/c56d00ee008007ad2a9474ed3af24cc0a0b20ebf)
- [fix: simplify string based types](https://github.com/ipld/js-dag-ucan/pull/50/commits/3686ebdc8e5991a2a2d2e81eba9d4361f918278d)
  - JWT<C> its just a string and phantom to retain C
  - SignPayload removed, not really useful
  - Block removed not used anywhere
  - New `UCANBlock` type, if we make the `Capability` generic an array we can merge this type with ucanto transport block
  - DID is now DIDString
  - Authority is now DIDVerifier
  - Some more tweaks and fixes for new TS version

closes https://github.com/ipld/js-dag-ucan/issues/49

### CODE REVIEW

**Review by Gozala — CHANGES_REQUESTED** [review]
Thanks @hugomrdias I'm mostly onboard with all the changes here, even if I don't like some of the new names.

I do however disagree with changes in regards to CID / Link stuff. It took me a lot of time and effort to introduce changes that would move from `CID` class to `Link` interface in multiformats and changes here seem to go into opposite direction. Can we please not do that, I know it causes some pain right now but as soon as new multiformats is out it will be all nice & pretty. We would also be able to remove some of the local types here in favor of ones in multiformats.

File: test/lib.spec.js
Comment (hugomrdias) [reply test/lib.spec.js:887]: i had to change the function signature/types a bit because of a new ts version. No need to add another test we would be just testing multiformat itself or a random function call.

File: package.json
Comment (hugomrdias) [reply package.json:56]: give me links please, i would love to but what was there before didnt work.

**Comment by hugomrdias** [comment]
in the last commit i changed DID back to being a string a renamed DID to Agent. Identity is still there because removing it breaks ucanto. We can remove it later after upgraded ucanto

File: package.json
Comment (Gozala) [reply package.json:56]: https://www.typescriptlang.org/docs/handbook/esm-node.html

File: src/ucan.ts
Comment (Gozala) [inline src/ucan.ts:175]: Nit: You could leave out version as `1` is a default.

```suggestion
> = IPLDLink<Model<Cap>, typeof CBOR_CODE, Alg> | IPLDLink<JWT<Cap>,typeof RAW_CODE, Alg>
```

File: src/ucan.ts
Comment (Gozala) [inline src/ucan.ts:269]: This was used in bunch of places & also a reason why I could not extend `CID` class because it's marked private there and that caused TS to complain.

I don't mind adding `//@ts-expect-error` in those places if this is really important, but just want to make sure you're aware of this.

**Review by Gozala — APPROVED** [review]
Pointed out few things, but I leave it up to you to decide what to do about them. Feel free to land as is or address some of those things and then land, I don't think we need another round of review here.

If you want to rename `Agent` to something else before landing that's cool with me as well.

File: package.json
Comment (hugomrdias) [reply package.json:56]: tried to do this but we cant change `module` in tsconfig yet it breaks other dep imports ...

**Pull request merged**
